04_flora_cnell/src/data_utils.R

```diff
@@ -0,0 +1,38 @@
+get_npn_layer <- function(layer_name, npn_layers, layer_date){
+  
+  layer_id <- npn_layers %>% 
+    filter(title == layer_name) %>%
+    pull(name)
+
+  if(!is.null(layer_date)){
+    label_date <- 'avg'
+  } else{
+    label_date <- layer_date
+  }
+  file_out <- gsub(':', '_', sprintf('out/%s_%s.tiff', layer_id, label_date))
+  npn_download_geospatial(coverage_id = layer_id,
+                          date = layer_date,
+                          format = 'geotiff',
+                          output_path = file_out
+  )
+  
+  return(file_out)
+  
+}
+prep_plot_df <- function(current_timing, proj){
+  anomaly_rast <- rast(current_timing) %>%
+    project(y = proj)
+  
+  # round to day
+  values(anomaly_rast) <- round(values(anomaly_rast), 0)
```

Comment:
Hitting an error here:
```r
Error : unable to find an inherited method for function 'values' for signature '"SpatRaster"'
```

Reply:
I updated my `raster` and `terra` packages and it ran, but with a bunch of warnings:
```r
Error in x$.self$finalize() : attempt to apply non-function
In addition: There were 12 warnings (use warnings() to see them)
Error in x$.self$finalize() : attempt to apply non-function
Error in (function (x)  : attempt to apply non-function
Error in (function (x)  : attempt to apply non-function
Error in (function (x)  : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in (function (x)  : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
Error in (function (x)  : attempt to apply non-function
Error in x$.self$finalize() : attempt to apply non-function
```

Reply:
Did updating `terra` resolve the `values` issue?

Reply:
These error messages are just messages and do not have any known consequences. Related to `rast`